### ISSUE TITLE
Cannot find DisplayBannerAndLaunch.bat 

### ISSUE THREAD

**Issue by twitchplaystrading** [IbcAlpha/IBC#99]
Hi,

I've been trying to use IBController which is outdated on both windows 10 and mac mojave with no avail. This time, i've completely reinstalled my windows and trying to get IBC to work. However, no matter which OS i'm running, i'm always getting the "cannot find DisplayBannerAndLaunch.bat" error as shown below. 
On macOS, i've even tried to change the original scripts to locate the DisplayBannerAndLaunch.sh with an absolute path but with no avail (IBController). I was hoping IBC would work but i guess not.. 

My IBC folder and config.ini is located in the default location like it was stated on the guide. 

Your help will be greatly appreciated.

![error 1](https://user-images.githubusercontent.com/74666134/99542976-fd665300-29f5-11eb-8706-2153899dc8c1.png)

**Comment by rlktradewright** [comment]
IBC and IBController are not the same thing. IBController is a discontinued project that is now completely unmaintained. The owner of that project should really get rid of it completely to avoid this sort of confusion, but has chosen not to do so.

IBC was forked from IBController several years ago and is fully maintained on Windows, Linux and macOS. 

You get IBC from here:

https://github.com/IbcAlpha/IBC

Note that you CANNOT convert an IBController installation to an IBC installation. You have to start from scratch with IBC. 

So start again with IBC, download it and read the User Guide - don't make any assumptions about how to set up IBC based on what you did with IBController: again, _they are different products_.

**Comment by twitchplaystrading** [comment]
> IBC and IBController are not the same thing. IBController is a discontinued project that is now completely unmaintained. The owner of that project should really get rid of it completely to avoid this sort of confusion, but has chosen not to do so.
> 
> IBC was forked from IBController several years ago and is fully maintained on Windows, Linux and macOS.
> 
> You get IBC from here:
> 
> https://github.com/IbcAlpha/IBC
> 
> Note that you CANNOT convert an IBController installation to an IBC installation. You have to start from scratch with IBC.
> 
> So start again with IBC, download it and read the User Guide - don't make any assumptions about how to set up IBC based on what you did with IBController: again, _they are different products_.

Hi Richard,

Thank you for the prompt reply but i am aware that the product discussed here is solely IBC (have read the ‘readme’). I was just mentioning IBController because the problem is simply the same: can’t find the DisplayBannerAndLaunch.bat.

I repeat: i am getting an persistent error that says it can’t find the DisplayBannerAndLaunch.bat whenever i run the scripts (either TWS or gateway).

**Comment by rlktradewright** [comment]
Ok, in that case attach your IBC log file to your reply so I can see what's going on.  (Please attach, do not quote it inline.)

**Comment by twitchplaystrading** [comment]
Sorry for keeping you waiting. As for now, I've solved the problems on macOS Mojave.
There were 3 types of them. 1) ibgateway-${tws_version}.vmoptions wasn't created so I've copied and renamed the 'tws.option' located in ${tws_path}. 
2) had to copy the '.install4j' file to the non-existent folder 'IB Gateway ${tws_version}. 
3) had to copy the 'jars' folder from 'Jts' to 'IB Gateway ${tws-version}'.

I will try it out again when I'm on Windows 10 and tell you whether the problem persists.

**Comment by rlktradewright** [comment]
There are two installers from IB on macOS (and the same is true for other platforms): one is for running TWS, the other is for the Gateway. They both install exactly the same jar files, but in different locations and with different scripts and different icons. 

(And in fact the Windows and Linux installers also install exactly the same jars: the only differences are where they are placed, the scripts to run them, and the icons that appear on the desktop.)

If you install from the TWS installer on macOS, the `IB Gateway ${tws_version}` folder is not created, and neither is `ibgateway-${tws_version}.vmoptions` (as you appear to have discovered).

But the IBC scripts are aware of this, and they will use whichever setup you have installed. So if you install from the TWS installer, and then you run gatewaystartmacos.sh, the script will simply use the relevant files and folders for the TWS installation. And vice versa: install from the Gateway installer and run twsstartmacos.sh, and it will use the files and folders for the Gateway installation.  It makes no difference at all as far as IBC is concerned.

So if you're needing to fiddle around creating 'missing' folders and files, it's entirely because you have messed something up. The IBC scripts have been carefully developed to make installation as simple as possible. If you follow the instructions in the User Guide, the ONLY files you might have to change are the relevant start scripts, ie twsstartmocs.sh or gatewaystartmacos.sh. And the ONLY things in those files that are likely to need changing are the major version number setting and the config.ini location setting. It really couldn't be much easier. 

So rather than try to 'fix' it, I suggest you simply delete your IBC folder, uninstall TWS/Gateway, and start again.

It literally takes less than 5 minutes to install and configure TWS/Gateway and IBC to get to a properly running IBC. (Longer the first time because you have to read the User Guide and follow the instructions.)

Oh, and regarding the permissions, on page 5 of the user guide it says:

```
On Unix:
* unpack the ZIP file using a command similar to this:
sudo unzip ~/Downloads/IBCLinux-3.6.0.zip -d \
/opt/ibc
* now make sure all the script files are executable:
cd /opt/ibc
sudo chmod o+x *.sh */*.sh
```
So RTFM!!!

Sorry if I sound a bit exasperated here, but that's because these sort of 'issues' take time to respond to that I'd rather spend on something else.

**Comment by twitchplaystrading** [comment]
Thanks for the feedback. I.. don't know what to say to make you feel any better. I've tried deleting and starting over but the issues kept happening on my mac. I'm not saying IBC is broken or anything and i'm by no means an expert on this matter but i just decided to figure it out given that's the issues are likely caused by my PC/mac not set correctly. I'm sorry if i sounded like i was undermining IBC or disrespecting in any way. 

Anyways, i've tried running IBC on Win 10 after deleting everything and starting over and i'm facing these issues. Attached the logs.

For TWS - 
[IBC-_TWS-981_FRIDAY.txt](https://github.com/IbcAlpha/IBC/files/5573074/IBC-_TWS-981_FRIDAY.txt)

For GateWay - 
[IBC-_GATEWAY-981_FRIDAY.txt](https://github.com/IbcAlpha/IBC/files/5573075/IBC-_GATEWAY-981_FRIDAY.txt)

**Comment by rlktradewright** [comment]
Quite right, IBC isn't broken!

What I can see immediately from your TWS log file is that you don't have the `IBC.jar` and `version` files in your `C:\IBC` folder.

I don't understand how this can be if you've followed these instructions on page 5 of the User Guide:

```
On Windows:

* create the folder where you want to install IBC, if it doesn’t already exist. As noted above (see Default Paths)
this is normally C:\IBC but it can be anywhere you like

* locate the downloaded ZIP file using File Explorer (Windows Explorer on Windows 7 and earlier). Windows
treats ZIP files like an ordinary folder, so you can see its contents the same way as any other folder

* select all the files and folders and drag them into your installation folder
```

And what I can see immediately from your Gateway log file is that your `C:\Jts\ibgateway\981\ibgateway.vmoptions` file is completely wrong. And I don't see how that can happen if you did what I suggested and uninstalled and the reinstalled Gateway.

So let me try again: please do the following:

1. Uninstall TWS 981 and Gateway 981 (or whichever one you have).

2. Just for good measure delete the `C:\Jts\ibgateway\981` and/or `C:\Jts\978` folders to make sure there is nothing left of the TWS/Gateway 981 installation (previous versions are not relevant).

3. Delete your `C:\IBC folder`. You don't need to delete your `C:\Users\alex\Documents\IBC\config.ini file`.

4. Install `ibgateway-latest-standalone-windows-x64.exe` or `tws-latest-standalone-windows-x64` - it doesn't matter which. The latest version of these files is 9.81.2.4 (you can see this in the Details tab of the file properties), but earlier versions of 981 will probably work just as well.

5. Download the `IBCWin-3.8.4-beta.2.zip` file from `https://github.com/IbcAlpha/IBC/releases/tag/3.8.4-beta.2`.

6. Extract the contents of the zip file into `C:\IBC`, as per the instructions quoted above (or right click on the .zip file and select 'Extract All...').

7. Edit `C:\IBC\StartGateway.bat`: change the `TWS_MAJOR_VRSN` setting to 981.

8. Run  `C:\IBC\StartGateway.bat`: voila! It just works....

**Comment by twitchplaystrading** [comment]
Thank you Richard. The problem has been resolved. Therefore, i will close the ticket.
